content/posts/2025-01-30-ebpf-flows-filtering.md

```diff
@@ -0,0 +1,228 @@
+
+The Following configuration example shows some of the possible filtering options with multi rules filters
+
+```yaml
+agent:
+    type: eBPF
+    ebpf:
+      flowFilter:
+        enable: true
+        rules:
+        - action: Accept
+          cidr: 10.128.0.0/24
+          peerCIDR: 10.129.0.0/24
+          ports: 443,6443
+          protocol: TCP
+          sampling: 10
+        - action: Accept
+          cidr: 10.129.0.1/24
+          ports: 53
+          protocol: UDP
+          sampling: 20
+        - action: Reject
+          tcpFlags: "SYN"
+          cidr: 10.130.0.0/24
+          protocol: TCP
+          sourcePorts: 80-100
+        - action: Accept
+          cidr: 172.30.0.0/16
+          protocol: SCTP
+          pktDrops: true
+        - action: Reject
+          cidr: 8.8.8.8/32
+          protocol: ICMP
+          icmpType: 8 // ICMP Echo request packet
+```
+2️⃣ Packet Inspection – Extract relevant packet attributes within an eBPF program.
+
+3️⃣ Early Flow Filtering –
+Skip or Allow packets based on predefined conditions before doing further ebpf packets processing.
+The Following Diagram shows how eBPF filtering is done
+
+<img src="{page.image('ebpf-flows-filtering/ebpf-flows-filtering.png')}" alt="eBPF Flows Filtering Processing">
+
+Note:
+- The maximum number of eBPF flow filter rules is limited to 16 to ensure conservative memory
+usage when this feature is enabled.
+
+- rules with duplicate CIDRs are rejected via validation webhook.
+
+- Both IPv4 and IPv6 formats are supported.
+
+- In case users wanted to match on any CIDR, default Null CIDR `0.0.0.0/0` can be used.
+
+## Key Use Cases
+
+🚀 Reducing Observability Overhead – Avoid logging irrelevant flows in high-traffic Kubernetes clusters.
+
+🔐 Security Filtering – Focus on anomalous or suspicious traffic while ignoring normal flows.
+
+🌐 Network Performance Monitoring – Capture only high-latency or dropped-packet flows for troubleshooting.
+
+### Filter EastWest and NorthSouth flows
+
+Let's allow serviceIP 172.30.100.64:80 with sample 10
+and allow pods traffic between subnet 10.128.0.0/16 and 10.129.0.0/16 with sampling 20 and reject everything else
+
+```yaml
+agent:
+    type: eBPF
+    ebpf:
+      flowFilter:
+        enable: true
+        rules:
+        - action: Accept
+          cidr: 172.30.0.0/16
+          sampling: 10
+        - action: Accept
+          cidr: 10.128.0.0/16
+          peerCIDR: 10.129.0.0/16
+          sampling: 20
+        - action: Reject
+          cidr: 0.0.0.0/0
+``` 
+
+<img src="{page.image('ebpf-flows-filtering/ebpf-svc-and-pods-flows.png')}" alt="eBPF Flows Filtering Kubernetes NorthSouth and EastWest Flows">
+
+### Filter flows with packet drops
+
+let's filter any kubernetes service flows with a packet drop and reject everything else,
+please note for this use case to work need to enable `PacketDrop` feature which requires `privileged` flag
+to be set to `true` as shows in the following sample configuration
```

Comment:
Can be updated to `Please not for this use case, we need to enable PacketDrop feature with eBPF in privileged mode as shown in the below configuration`